coordinator/main.go

```diff
@@ -130,6 +139,7 @@ func run() (retErr error) {
 			logger.Error("Serving Coordinator API", "err", err)
 			return fmt.Errorf("serving Coordinator API: %w", err)
 		}
+		*userapiStarted = true
```

Comment:
I fear that this is too late - `Serve` returns if there's a fatal error while serving, or the server is closed. Needs to move to before the `Serve` call. Same below.